### ISSUE TITLE
Impossible to skip PIN entering

### ISSUE THREAD

**Issue by microshine** [PeculiarVentures/fortify-tools#62]
The application shows a PIN dialog if the token requires login. On Cancel button click it prompts PIN entering again and again
![image](https://user-images.githubusercontent.com/7333669/119856075-1d358100-bf1b-11eb-8491-b246664a705c.png)
![image](https://user-images.githubusercontent.com/7333669/119856201-363e3200-bf1b-11eb-8a2f-63632df7c953.png)
![image](https://user-images.githubusercontent.com/7333669/119856238-3c341300-bf1b-11eb-98b3-535b1b6cec4f.png)

**Comment by alexgbv** [comment]
@microshine It seems we have a contradiction between layout and application, I think need to align
![image](https://user-images.githubusercontent.com/39085824/120204845-f08eab80-c231-11eb-8838-a12fb2131631.png)

-----

I suppose when click "Deny" we don't need to show an additional view. It should turn you to homepage. What do you think?

**Comment by microshine** [comment]
Agree. Looks like Fortify throes exception on PIN Denying. Mane we need to send a special code error to catch that behaviour

**Comment by donskov** [comment]
@alexgbv Also, we need to add a `Logout` button and show it only for smart cards and after successful authentication. After a click, we should trigger re-authentication for a smart card.

**Comment by alexgbv** [comment]
@donskov If I correctly understand you we need some indicator of smartcard connection in cert viewer, and also it must have the option of logout. Is it around?

**Comment by donskov** [comment]
@alexgbv We need to add the button `login/logout` for the removable provider (smart-card).

I think it's related for `tools` only for now.

**Comment by alexgbv** [comment]
It is for logout
![image](https://user-images.githubusercontent.com/39085824/121521113-58947d00-c9fc-11eb-9f7d-aff6e9567a8f.png)

I think we should deselect token from providers if logged out instead show empty page. IMHO it will be more natural behaviour than show nothing. And to login - select it in providers again.

**Comment by rossinicolas** [comment]
Good Morning Dears, i think that the upgrade requested from my users is related to this issue. They told me if there are any way to grant greater security to PeculiarFortifyCertificates web component at moment of show and choose certificate to sign because if token card is plugin and they leave from your desk for a few minutes, other guys could be use it to sign. May be using a PIN or pass could be a solution for us.
Thanks.

**Comment by rmhrisk** [comment]
@rossinicolas this should really be in a separate bug. That said pin cashing / policy is managed by smart card middleware and not fortify. Fortify could implement its own policy that is additive but it can not replace what is there. Please file a bug requesting this feature.
